### PR TITLE
Fix unresolved doxygen for N-dimensional FFT APIs

### PR DESCRIPTION
**Pull request by yasahi-hpc**
This PR aims at fixing docstrings bugs from #141 
As a temporary solution, we introduce a macro for doxygen which can be resolved. See https://github.com/kokkos/kokkos-kernels/pull/1319 for detail.

### CODE REVIEW

File: fft/src/KokkosFFT_Transform.hpp [hunk 1]
```diff
@@ -469,11 +469,16 @@ void irfft2(const ExecutionSpace& exec_space, const InViewType& in,
 /// \param exec_space [in] Kokkos execution space
 /// \param in [in] Input data (complex)
 /// \param out [out] Ouput data (complex)
-/// \param axes [in] Axes over which FFT is performed (optional)
+/// \param axes [in] Axes over which FFT is performed (default, all axes)
```
Comment (tpadioleau) [inline fft/src/KokkosFFT_Transform.hpp:472]: I think the word `default` is better than `optional`. I mean one cannot only specify the parameter `s` without specifying `axes` and `norm`. The full set `axes`,`norm` and `s` can be interpreted as optional but not really separately ?

**Review by tpadioleau — APPROVED** [review]
LGTM

File: fft/src/KokkosFFT_Transform.hpp (hunk 1 above)
Comment (tpadioleau) [reply fft/src/KokkosFFT_Transform.hpp:472]: I forgot a concluding remark, should we consider updating all `(optional)` to `(default, ...)` ?

**Comment by yasahi-hpc** [comment]
@tpadioleau Thank you for your review.

File: fft/src/KokkosFFT_Transform.hpp (hunk 1 above)
Comment (tpadioleau) [reply fft/src/KokkosFFT_Transform.hpp:472]: sure

**Pull request merged**